src/ast.ts

```diff
@@ -174,6 +174,30 @@ export abstract class Node {
     return new TypeParameterNode(name, extendsType, defaultType, range);
   }
 
+  static createCompoundNamedType(
+    identifiers: IdentifierExpression[],
+    typeParameters: TypeNode[] | null,
+    isNullable: bool,
+    range: Range
+  ): NamedTypeNode {
+    let length = identifiers.length;
+    if (length == 0) throw new Error(); // how should errors be handled?
```

Comment:
If it was purely internal, I'd recommend an `assert`, but I think throwing is fine here since this is kinda a user-facing API when modifying the AST in transforms. Perhaps `Error("identifiers must not be empty")`.